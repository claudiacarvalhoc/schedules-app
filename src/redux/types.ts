export const FETCH_SCHEDULE_SUCCESS = 'FETCH_SCHEDULE_SUCCESS';
export const FETCH_SCHEDULE_FAILURE = 'FETCH_SCHEDULE_FAILURE';

export const FETCH_SCHEDULE_LOGS_SUCCESS = 'FETCH_SCHEDULE_LOGS_SUCCESS';
export const FETCH_SCHEDULE_LOGS_FAILURE = 'FETCH_SCHEDULE_LOGS_FAILURE';

export const REQUEST_RETIRE_SCHEDULE = 'REQUEST_RETIRE_SCHEDULE';
export const REQUEST_UNRETIRE_SCHEDULE = 'REQUEST_UNRETIRE_SCHEDULE';
export const UPDATE_SELECTED_SCHEDULE = 'UPDATE_SELECTED_SCHEDULE';