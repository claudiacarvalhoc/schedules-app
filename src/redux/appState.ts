import { DefaultRootState } from "react-redux";

/**
 * The App Status
 */
export interface AppState extends DefaultRootState {
    status: AppStatusSectionsState;
    selectedScheduleId?: number;
    schedules: ScheduleState[];
    scheduleLogs: ScheduleLogState[];
    texts: {
        header: HeaderTextStatus;
        schedules: ScheduleTextState;
    }
}

/**
 * The status to fetch data per section
 */
export interface AppStatusSectionsState {
    schedules: StatusState;
    scheduleLogs: StatusState;
}

/**
 * The Boot Status is defined for options: Loading / Success / Failure.
 */
export enum StatusState {
    Loading,
    Success,
    Failure
}

/**
 * The Schedule interface.
 */
export interface ScheduleState {
    id: number;
    avatar: AvatarState;
    name: string;
    description: string;
    isRetired: boolean;
    tasksCount: number;
    startPoint: Date;
    endPoint: Date;
    dayOfWeek: number;
    dayOfMonth: number;
    startDate: Date;
    endDate: Date;
    schedules?: IntervalTypeState;
    timePeriod?: number;
}

/**
 * The Avatar Identification
 */
export interface AvatarState {
    letter: string;
    color: string;
}

/**
 * The Schedule Interval Types
 */
export enum IntervalTypeState {
    Once,
    Year,
    Week,
    Hour,
    Minute,
    Second
}

/**
 * The Schedule Logs interface.
 */
export interface ScheduleLogState {
    id: number;
    startTime: Date;
    endTime: Date;
    status: LogStatusState;
    serverName: string;
    scheduleId: number;
}

/**
 * The Schedule Logs Status
 */
export enum LogStatusState {
    Terminated,
    Pending,
    Completed,
    Exception,
    Running
}

/**
 * Texts used on Header
 * Note: This allow us to add localisation feature easily
 */
export interface HeaderTextStatus {
    titleText: string;
}

/**
 * Texts used on Schedule
 * Note: This allow us to add localisation feature easily
 */
export interface ScheduleTextState {
    buttonRetireText: string;
    buttonUnretireText: string;
    errorMessageText: string;
    buttonRetryText: string;
    emptyMessageText: string;
}
