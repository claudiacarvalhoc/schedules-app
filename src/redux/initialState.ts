import { AppState, Status } from "./appState";

/**
 * This is the initial state of redux.
 */
export const appInitialState: AppState = {
    status: {
        schedules: Status.Loading,
        scheduleLogs: Status.Loading,
    },
    selectedScheduleId: undefined,
    schedules: [],
    scheduleLogs: [],
};
