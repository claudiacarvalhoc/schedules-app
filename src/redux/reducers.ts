import { Reducer, combineReducers } from 'redux';
import { AppState } from './appState';
import { appReducer } from './app/reducer';

export const rootReducer = combineReducers({
    app: appReducer as Reducer<AppState>,
});

export type RootState = ReturnType<typeof rootReducer>;
export type DefaultRootState = ReturnType<typeof rootReducer>;