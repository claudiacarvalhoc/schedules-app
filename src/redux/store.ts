import { AnyAction } from 'redux';
import { AppState } from './appState';
import { ThunkAction, ThunkDispatch } from 'redux-thunk';
import { applyMiddleware, compose, createStore } from 'redux';
import { createLogger } from 'redux-logger';
import { rootReducer } from './reducers';
import thunk from 'redux-thunk';

export type AppThunk<R> = ThunkAction<R, AppState, unknown, AnyAction>;
export type AppDispatch = ThunkDispatch<AppState, unknown, AnyAction>;


export const configureStore = (initialState: AppState) => {
    const logger = createLogger();
    const composeEnhancers = (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
    const middleware = [thunk, logger];

    const store = createStore(
        rootReducer,
        {
            app: initialState,
        },
        composeEnhancers(applyMiddleware(...middleware)),
    );

    return store;
};

export default configureStore;
