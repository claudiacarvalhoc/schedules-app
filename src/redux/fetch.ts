import { AppDispatch } from './store';

export const getSchedules = (): ((dispatch: AppDispatch) => Promise<void>) => {
    return dispatch => {
        // TODO : ....
        return Promise.resolve();
    };
};

export const getScheduleLogs = (): ((dispatch: AppDispatch) => Promise<void>) => {
    return dispatch => {
        // TODO : ....
        return Promise.resolve();
    };
};
