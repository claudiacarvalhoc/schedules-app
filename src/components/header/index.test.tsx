import React from 'react';
import Header from './index';
import Enzyme, { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { appInitialState } from '../../redux/initialState';
import { HeaderText } from '../../redux/appState';

Enzyme.configure({ adapter: new Adapter() });

const getMountWrapper = (headerText: HeaderText) => {
    const store = configureMockStore()({
      app: {
        ...appInitialState,
        texts: {
            ...appInitialState.texts,
            header: headerText
        }
      }
    });
    return mount(
        <Provider store={store}>
            <Header />
        </Provider>
    );
  };

describe('<Header />', () => {
    it('render correctly', () => {
        const props: HeaderText = {
            titleText: 'Schedule',
        };
        const component = getMountWrapper(props);
        expect(toJson(component)).toMatchSnapshot();
    });
});
