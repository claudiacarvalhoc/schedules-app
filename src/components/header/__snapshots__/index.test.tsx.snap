// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Header)>
    <Header
      search={[Function]}
      title="Schedule"
    >
      <div
        className="container"
      >
        <WithStyles(ForwardRef(AppBar))
          position="static"
        >
          <ForwardRef(AppBar)
            classes={
              Object {
                "colorDefault": "MuiAppBar-colorDefault",
                "colorInherit": "MuiAppBar-colorInherit",
                "colorPrimary": "MuiAppBar-colorPrimary",
                "colorSecondary": "MuiAppBar-colorSecondary",
                "colorTransparent": "MuiAppBar-colorTransparent",
                "positionAbsolute": "MuiAppBar-positionAbsolute",
                "positionFixed": "MuiAppBar-positionFixed",
                "positionRelative": "MuiAppBar-positionRelative",
                "positionStatic": "MuiAppBar-positionStatic",
                "positionSticky": "MuiAppBar-positionSticky",
                "root": "MuiAppBar-root",
              }
            }
            position="static"
          >
            <WithStyles(ForwardRef(Paper))
              className="MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary"
              component="header"
              elevation={4}
              square={true}
            >
              <ForwardRef(Paper)
                className="MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary"
                classes={
                  Object {
                    "elevation0": "MuiPaper-elevation0",
                    "elevation1": "MuiPaper-elevation1",
                    "elevation10": "MuiPaper-elevation10",
                    "elevation11": "MuiPaper-elevation11",
                    "elevation12": "MuiPaper-elevation12",
                    "elevation13": "MuiPaper-elevation13",
                    "elevation14": "MuiPaper-elevation14",
                    "elevation15": "MuiPaper-elevation15",
                    "elevation16": "MuiPaper-elevation16",
                    "elevation17": "MuiPaper-elevation17",
                    "elevation18": "MuiPaper-elevation18",
                    "elevation19": "MuiPaper-elevation19",
                    "elevation2": "MuiPaper-elevation2",
                    "elevation20": "MuiPaper-elevation20",
                    "elevation21": "MuiPaper-elevation21",
                    "elevation22": "MuiPaper-elevation22",
                    "elevation23": "MuiPaper-elevation23",
                    "elevation24": "MuiPaper-elevation24",
                    "elevation3": "MuiPaper-elevation3",
                    "elevation4": "MuiPaper-elevation4",
                    "elevation5": "MuiPaper-elevation5",
                    "elevation6": "MuiPaper-elevation6",
                    "elevation7": "MuiPaper-elevation7",
                    "elevation8": "MuiPaper-elevation8",
                    "elevation9": "MuiPaper-elevation9",
                    "outlined": "MuiPaper-outlined",
                    "root": "MuiPaper-root",
                    "rounded": "MuiPaper-rounded",
                  }
                }
                component="header"
                elevation={4}
                square={true}
              >
                <header
                  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
                >
                  <WithStyles(ForwardRef(Toolbar))
                    className="bar"
                  >
                    <ForwardRef(Toolbar)
                      className="bar"
                      classes={
                        Object {
                          "dense": "MuiToolbar-dense",
                          "gutters": "MuiToolbar-gutters",
                          "regular": "MuiToolbar-regular",
                          "root": "MuiToolbar-root",
                        }
                      }
                    >
                      <div
                        className="MuiToolbar-root MuiToolbar-regular bar MuiToolbar-gutters"
                      >
                        <WithStyles(ForwardRef(Typography))
                          variant="h6"
                        >
                          <ForwardRef(Typography)
                            classes={
                              Object {
                                "alignCenter": "MuiTypography-alignCenter",
                                "alignJustify": "MuiTypography-alignJustify",
                                "alignLeft": "MuiTypography-alignLeft",
                                "alignRight": "MuiTypography-alignRight",
                                "body1": "MuiTypography-body1",
                                "body2": "MuiTypography-body2",
                                "button": "MuiTypography-button",
                                "caption": "MuiTypography-caption",
                                "colorError": "MuiTypography-colorError",
                                "colorInherit": "MuiTypography-colorInherit",
                                "colorPrimary": "MuiTypography-colorPrimary",
                                "colorSecondary": "MuiTypography-colorSecondary",
                                "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                "displayBlock": "MuiTypography-displayBlock",
                                "displayInline": "MuiTypography-displayInline",
                                "gutterBottom": "MuiTypography-gutterBottom",
                                "h1": "MuiTypography-h1",
                                "h2": "MuiTypography-h2",
                                "h3": "MuiTypography-h3",
                                "h4": "MuiTypography-h4",
                                "h5": "MuiTypography-h5",
                                "h6": "MuiTypography-h6",
                                "noWrap": "MuiTypography-noWrap",
                                "overline": "MuiTypography-overline",
                                "paragraph": "MuiTypography-paragraph",
                                "root": "MuiTypography-root",
                                "srOnly": "MuiTypography-srOnly",
                                "subtitle1": "MuiTypography-subtitle1",
                                "subtitle2": "MuiTypography-subtitle2",
                              }
                            }
                            variant="h6"
                          >
                            <h6
                              className="MuiTypography-root MuiTypography-h6"
                            >
                              Schedule
                            </h6>
                          </ForwardRef(Typography)>
                        </WithStyles(ForwardRef(Typography))>
                        <div
                          className="search"
                        >
                          <WithStyles(ForwardRef(InputBase))
                            classes={
                              Object {
                                "input": "inputInput",
                                "root": "inputRoot",
                              }
                            }
                            inputProps={
                              Object {
                                "aria-label": "search",
                              }
                            }
                            onChange={[Function]}
                            placeholder="Search…"
                          >
                            <ForwardRef(InputBase)
                              classes={
                                Object {
                                  "adornedEnd": "MuiInputBase-adornedEnd",
                                  "adornedStart": "MuiInputBase-adornedStart",
                                  "colorSecondary": "MuiInputBase-colorSecondary",
                                  "disabled": "Mui-disabled",
                                  "error": "Mui-error",
                                  "focused": "Mui-focused",
                                  "formControl": "MuiInputBase-formControl",
                                  "fullWidth": "MuiInputBase-fullWidth",
                                  "input": "MuiInputBase-input inputInput",
                                  "inputAdornedEnd": "MuiInputBase-inputAdornedEnd",
                                  "inputAdornedStart": "MuiInputBase-inputAdornedStart",
                                  "inputHiddenLabel": "MuiInputBase-inputHiddenLabel",
                                  "inputMarginDense": "MuiInputBase-inputMarginDense",
                                  "inputMultiline": "MuiInputBase-inputMultiline",
                                  "inputTypeSearch": "MuiInputBase-inputTypeSearch",
                                  "marginDense": "MuiInputBase-marginDense",
                                  "multiline": "MuiInputBase-multiline",
                                  "root": "MuiInputBase-root inputRoot",
                                }
                              }
                              inputProps={
                                Object {
                                  "aria-label": "search",
                                }
                              }
                              onChange={[Function]}
                              placeholder="Search…"
                            >
                              <div
                                className="MuiInputBase-root inputRoot"
                                onClick={[Function]}
                              >
                                <input
                                  aria-label="search"
                                  className="MuiInputBase-input inputInput"
                                  onAnimationStart={[Function]}
                                  onBlur={[Function]}
                                  onChange={[Function]}
                                  onFocus={[Function]}
                                  placeholder="Search…"
                                  type="text"
                                />
                              </div>
                            </ForwardRef(InputBase)>
                          </WithStyles(ForwardRef(InputBase))>
                        </div>
                        <ForwardRef(MenuIcon)>
                          <WithStyles(ForwardRef(SvgIcon))>
                            <ForwardRef(SvgIcon)
                              classes={
                                Object {
                                  "colorAction": "MuiSvgIcon-colorAction",
                                  "colorDisabled": "MuiSvgIcon-colorDisabled",
                                  "colorError": "MuiSvgIcon-colorError",
                                  "colorPrimary": "MuiSvgIcon-colorPrimary",
                                  "colorSecondary": "MuiSvgIcon-colorSecondary",
                                  "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                  "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                  "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                  "root": "MuiSvgIcon-root",
                                }
                              }
                            >
                              <svg
                                aria-hidden={true}
                                className="MuiSvgIcon-root"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                                />
                              </svg>
                            </ForwardRef(SvgIcon)>
                          </WithStyles(ForwardRef(SvgIcon))>
                        </ForwardRef(MenuIcon)>
                      </div>
                    </ForwardRef(Toolbar)>
                  </WithStyles(ForwardRef(Toolbar))>
                </header>
              </ForwardRef(Paper)>
            </WithStyles(ForwardRef(Paper))>
          </ForwardRef(AppBar)>
        </WithStyles(ForwardRef(AppBar))>
      </div>
    </Header>
  </Connect(Header)>
</Provider>
`;
